models/ed/R/write_restart.ED2.R

```diff
@@ -1,100 +1,90 @@
 #' @title Write ED2 restart file from SDA results
 #' 
-#' @author Alexey Shiklomanov
+#' @author Alexey Shiklomanov, Istem Fer
 #' @inheritParams PEcAn.ModelName::write_restart.ModelName
 #' @return TRUE if successful
 #' @export
-write_restart.ED2 <- function(outdir,
-                              runid,
-                              start.time,
-                              stop.time,
-                              settings,
-                              new.state) {
-
-  rundir <- settings$host$rundir
-  mod_outdir <- settings$host$outdir
-
-  sda_datestr <- strftime(start.time, "%Y-%m-%d-%H%M%S")
-  sda_suffix <- paste0("SDA.", sda_datestr)
-
-  # Get history restart file path
-  histfile <- get_restartfile.ED2(mod_outdir, runid, start.time)
-  if (is.null(histfile)) {
-    PEcAn.logger::logger.severe("Failed to find ED2 history restart file.")
-  }
-
+write_restart.ED2 <- function(outdir, runid, start.time, stop.time,
+                              settings, new.state, new.params, inputs, RENAME = TRUE) {
+  
+  restart <- new.params$restart
+  
+  old.state <- restart$oldstate # hack: this will probably change in the near future, it's currently just AbvGrndWood 
+  histfile  <- restart$histfile # Get history restart file path
+  restart   <- restart$restart
```

Comment:
I agree that it's a bit unintuitive.. But write_restart.ED2 (at least the way we do things currently) requires reading back the old state, and to read the old state we need to find out the histfile name again etc.. I thought it's easier to just pass them around and new.params was convenient

In SIPNET we were passing just params to update things deterministically but as with ED2 I can imagine other write_restarts might want things that are created/read in read_restarts? Maybe should change the name of `new.params` to something like more practical like `pass.to.wr`?

Reply:
"new parameters" and "other misc info that write restart needs" are categorically very different things, which is why I objected to passing things that are not parameters along with the parameter list. If there are other things that need to be passed then we should discuss how/where that occurs, rather than renaming `new.params` and muddling the development direction of joint SDA/PDA.

Reply:
I'd be down for that..I think something for "other misc info that write restart needs" will make life a lot easier